docs/codecs.rst

```diff
@@ -178,6 +178,35 @@ header. The format of the encoded buffer is defined in [BLOSC]_. The
 reference implementation is provided by the `c-blosc library
 <https://github.com/Blosc/c-blosc>`_.
 
+Endian
+------
+
+Codec URI:
+    https://purl.org/zarr/spec/codec/endian
+
+Encodes array elements using the specified endianness.
+
+Configuration parameters
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+endian:
+    Required.  A string equal to either ``"big"`` or ``"little"``.
+
+Format and algorithm
+~~~~~~~~~~~~~~~~~~~~
+
+Each element of the array is encoded using the specified endian variant of its
+default binary representation.  Array elements are encoded in lexicographical
+order.  For example, with ``endian`` specified as ``big``, the ``int32`` data
+type is encoded as a 4-byte big endian two's complement integer, and the
+``complex128`` data type is encoded as two consecutive 8-byte big endian IEEE
+754 binary64 values.
+
+.. note::
+
+   Single the default binary representation of all data types is little endian,
+   specifying this codec with ``endian`` equal to ``"little"`` is equivalent to
+   omitting this codec.
```

Comment:
If I understand this correctly, an implementation could also decide to return a big-endian representation, which reverts this logic, right?

Reply:
I would say that the in-memory representation used by a zarr implementation is out-of-scope for the specification.

But I should clarify that "equivalent to omitting this codec" is just because the default conversion from logical array to byte sequence uses the default binary representation, which is always little endian.  However, "equivalent to omitting this codec" is not the same as "no byte order conversion from the in-memory representation is performed"; indeed on a big endian machine the in-memory representation would likely be big endian and therefore if the endian codec is not specified with big endian, then a byte order conversion will be needed to convert to the binary representation.

Reply:
Great, thanks :+1: From my side this LGTM, when those clarifications are added.

Reply:
Fixed.